src/StatusQ/Controls/StatusBaseButton.qml

```diff
@@ -18,159 +18,119 @@ Rectangle {
+        default:
+            return 11
         }
     }
 
+    implicitWidth: contentItem.width + root.leftPadding + root.rightPadding
+    implicitHeight: contentItem.height + root.topPadding + root.bottomPadding
+    enabled: !loading
+    leftPadding: defaultLeftPadding
+    rightPadding: defaultRightPadding
+    topPadding: defaultTopPadding
+    bottomPadding: defaultBottomPadding
 
-    property real leftPadding: defaultLeftPadding
-    property real rightPadding: defaultRightPadding
-    property real topPadding: defaultTopPadding
-    property real bottomPadding: defaultBottomPadding
-
-    property color normalColor
-    property color hoverColor
-    property color disaledColor
-
-    property color textColor
-    property color disabledTextColor
-
-    signal pressed(var mouse)
-    signal released(var mouse)
-    signal clicked(var mouse)
-    signal pressAndHold(var mouse)
-
-    property bool highlighted: false
-
-
-    /// Implementation
-    implicitWidth: layout.width + statusBaseButton.leftPadding + statusBaseButton.rightPadding
-    implicitHeight: layout.height + statusBaseButton.topPadding + statusBaseButton.bottomPadding
-
-
-    radius: size !== StatusBaseButton.Size.Tiny ? 8 : 6
-
-    color: {
-        if (statusBaseButton.enabled)
-            return sensor.containsMouse || highlighted ? hoverColor
-                                                       : normalColor;
-        return disaledColor
+    background: Rectangle {
+        anchors.fill: parent
+        radius: root.size !== StatusBaseButton.Size.Tiny ? 8 : 6
+        border.color: root.borderColor
+        color: {
+            if (root.enabled)
+                return root.hovered || highlighted ? hoverColor : normalColor;
+            return disaledColor
+        }
+        MouseArea {
+            id: mouseArea
+            anchors.fill: parent
+            cursorShape: loading ? Qt.ArrowCursor : Qt.PointingHandCursor;
+            onPressed: mouse.accepted = false;
+        }
     }
 
     QtObject {
         id: d
-        readonly property color textColor: statusBaseButton.enabled ? statusBaseButton.textColor : statusBaseButton.disabledTextColor
+        readonly property color textColor: root.enabled ? root.textColor : root.disabledTextColor
     }
 
-    MouseArea {
-        id: sensor
-        anchors.fill: parent
-        cursorShape: loading ? Qt.ArrowCursor
-                             : Qt.PointingHandCursor
-
-        hoverEnabled: true
-        enabled: !loading && statusBaseButton.enabled
+    contentItem: Item {
+        id: contentItem
+        width: layout.width
+        height: layout.height
 
         Loader {
             anchors.centerIn: parent
             active: loading
             sourceComponent: StatusLoadingIndicator {
                 color: d.textColor
-            } // Indicator
-        } // Loader
+            }
+        }
 
         Row {
             id: layout
             anchors.centerIn: parent
```

Comment:
I guess it's the same since implicitWidth/implicitHeight are calculated based on the content item's dimensions plus the height, as an example here below the very left button has `leftPadding: 0` and the rest the default values:
<img width="392" alt="jj" src="https://user-images.githubusercontent.com/31625338/181736697-40c6f6f8-de17-49d2-a36a-ef0dc33e6f68.png">